### PR TITLE
Peaks.destroy

### PR DESCRIPTION
**Pull request by jdelStrother**
This branch adds a Peaks.destroy method, used to clean up event listeners on the window and the media element.   WDYT?

### CODE REVIEW

**Comment by chrisn** [comment]
Note: see also https://github.com/bbcrd/peaks.js/issues/69.

File: src/main/views/waveform.overview.js [hunk 1]
```diff
@@ -231,5 +231,10 @@ define([
     this.uiLayer.draw();
   };
 
+  WaveformOverview.prototype.destroy = function () {
+    this.stage.destroy();
+    this.state = null;
```
Comment (chrisn) [inline src/main/views/waveform.overview.js]: Do you mean `this.stage`?

File: src/main/views/waveform.zoomview.js [hunk 2]
```diff
@@ -400,5 +400,10 @@ define([
     this.updateZoomWaveform(this.frameOffset);
   };
 
+  WaveformZoomView.prototype.destroy = function () {
+    this.stage.destroy();
+    this.state = null;
```
Comment (chrisn) [inline src/main/views/waveform.zoomview.js]: `this.stage`?

File: src/main/views/waveform.overview.js (hunk 1 above)
Comment (jdelStrother) [inline src/main/views/waveform.overview.js]: Almost certainly, sorry.   I'll push a fix.

**Comment by chrisn** [comment]
Thanks again for your contributions. Could you tell us about the problems you're seeing that require this change, and any testing you've done to show that this PR fixes them (eg, memory profiling?).

**Comment by jdelStrother** [comment]
We're using this in a single-page React app, and we create & discard multiple peaks instances through the duration of the page.   The event listeners on the discarded ones would stick around though, so we'd quite often see the error "options.width should be a strictly positive integer value" during resizing the window, for example.   That error was the main impetus, rather than having problems with memory usage.

**Comment by chrisn** [comment]
Was the error repeatable? It'd be nice to have a test case around it.

**Comment by jdelStrother** [comment]
I’ll see what I can do.

**Comment by jdelStrother** [comment]
I managed to come up with a test for it... it's not pretty, but see what you think.   https://github.com/bbcrd/peaks.js/pull/150/files#diff-d83a089f2fdd993eaf02f77a31969e49R92

**Comment by chrisn** [comment]
Looks good to me, and tests OK locally in Chrome and Firefox. Is this ready to merge from your point of view?

**Comment by jdelStrother** [comment]
Yes please!

**Comment by oncletom** [comment]
Small question @jdelStrother: why do you use your own `listeners` array and `_addMediaListener` instead of relying on the [`EventEmitter2.removeAllListeners`](https://github.com/asyncly/EventEmitter2#emitterremovealllistenersevent) method – Peaks class extends EventEmitter2?

**Comment by jdelStrother** [comment]
Can you use eventemitter for binding native events?

**Comment by oncletom** [comment]
@jdelStrother hm, no and looking at the code, I don't think it will remove much of what you already wrote. It would be useful just a way to proxy the native media element events.

This is very good like it is at the moment, thanks for that :-)

**Pull request merged**